disco_aws_automation/disco_aws.py

```diff
@@ -308,21 +306,95 @@ def update_elb(self, hostclass, update_autoscaling=True, testing=False):
                 elb_subnets = self.get_subnets(elb_meta_network, hostclass)
                 subnet_ids = [subnet['SubnetId'] for subnet in elb_subnets]
 
-            elb_port = self.hostclass_option_default(hostclass, "elb_port", 80)
-            elb_protocol = self.hostclass_option_default(hostclass, "elb_protocol", None) or \
-                self._default_protocol_for_port(elb_port)
-            instance_port = int(self.hostclass_option_default(hostclass, "elb_instance_port", 80))
-            instance_protocol = self.hostclass_option_default(hostclass, "elb_instance_protocol", None) or \
-                self._default_protocol_for_port(instance_port)
+            def default_protocol_for_port(port):
+                """
+                Return the default protocl for the port
+                """
+                return {80: 'HTTP', 443: 'HTTPS'}.get(int(port), 'TCP')
+
+            def default_port_for_protocol(protocol):
+                """
+                Return the default port for the protocol
+                """
+                return {'HTTP': 80, 'HTTPS': 443}[protocol]
+
+            def unzip(things):
+                """
+                The inverse of zip
+                """
+                return zip(*things)
+
+            def zip_with_defaults(x_things, y_things, default_x, default_y):
+                """
+                Zip together two iterables, replacing missing values with the
+                result of calling the supplied default callable
+                """
+                return [
+                    (
+                        default_x(y) if x is None else x,
+                        default_y(x) if y is None else y,
+                    )
+                    for x, y in izip_longest(
+                        x_things,
+                        y_things,
+                        fillvalue=None
+                    )
+                ]
+
+            def list_from_hostclass_option(option):
+                """
+                Return a list of values from a hostclass option
+                """
+                values = self.hostclass_option_default(hostclass, option, '')
+
+                return values.split(',') if values else []
+
+            elb_ports = [
+                int(port)
+                for port in list_from_hostclass_option(
+                    'elb_port'
+                )
+            ]
+            elb_protocols = list_from_hostclass_option('elb_protocol')
+            elb_protocols = [protocol.strip() for protocol in elb_protocols]
+            elb_ports, elb_protocols = unzip(
+                zip_with_defaults(
+                    elb_ports,
+                    elb_protocols,
+                    default_port_for_protocol,
+                    default_protocol_for_port
+                ) or [(80, 'HTTP')]
+            )
+
+            instance_ports = [
```

Comment:
Agreed, I was thinking of factoring this out, but I wanted a second opinion before I went ahead and did it.